components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs

```diff
@@ -58,6 +58,69 @@ public static class PublishedNodesEntryModelEx {
             return true;
         }
 
+        /// <summary>
+        /// filter model
+        /// </summary>
+        public static bool IsFiltered(this PublishedNodesEntryModel model, PublishedNodesEntryModel that) {
+
+            if (model == that) {
+                return true;
+            }
+
+            if (model == null || that == null) {
+                return false;
+            }
+
+            if (string.Compare(model.DataSetWriterGroup, that.DataSetWriterGroup, StringComparison.InvariantCulture) != 0 &&
+                that.DataSetWriterGroup != null) {
```

Comment:
I'm not sure if this is correct for all use-cases 
e.g.:
with
 model.DataSetWriterGroup = "1"
 that.DataSetWriterGroup = null 

the result will be true, meaning the fields are equal, which is not exactly right.

Reply:
This is used to filter the nodes on endpoint.
model is the actual entry, that.DataSetWriterGroup is the filtering mask. If that.DataSetWriterGroup is null it means that every DataSetWriterGroup is valid.
if we have 2 DataSetWriterGroup with same endpointUrl and just use the endpointUrl to filter the nodes, we get all nodes from both DataSetWriterGroup

Reply:
I'm not sure if I agree with this: 

> if we have 2 DataSetWriterGroup with same endpointUrl and just use the endpointUrl to filter the nodes, we get all nodes from both DataSetWriterGroup

I think we should not mix in the reply nodes from different DataSetWriterGroups. The problem is that the user will not be able to differentiate which node belongs in which group. If the request is ambiguous, we should just return an error so the user can refine it's request.

Reply:
fixed